### PR TITLE
Truncate patch summaries to db field size (512chr)

### PR DESCRIPTION
**Pull request by wstephenson**
Fixes https://bugzilla.suse.com/show_bug.cgi?id=1117190
Method taken from https://bugzilla.suse.com/show_bug.cgi?id=1049788

### CODE REVIEW

File: www/perl-lib/SMT/Patch.pm [hunk 1]
```diff
@@ -380,11 +380,16 @@ sub save
         $description = substr($description, 0, SMT::Utils::MYSQL_TEXT_TYPE_SIZE);
     }
 
-    my $sth = $dbh->prepare($sql);
+    my $summary = $self->summary();
+    if (length($summary) > SMT::Utils::PATCH_SUMMARY_SIZE_CHARS) {
+        $summary = substr($summary, 0, SMT::Utils::MYSQL_TEXT_TYPE_SIZE);
+    }
+
+     my $sth = $dbh->prepare($sql);
```
Comment (fschueller) [inline www/perl-lib/SMT/Patch.pm]: Is there a specific reason for indenting this line by one?

**Review by fschueller — APPROVED** [review]
To me it seems fine though otherwise, following the same logic as the fix for the description field.

File: www/perl-lib/SMT/Patch.pm (hunk 1 above)
Comment (wstephenson) [reply www/perl-lib/SMT/Patch.pm]: Unspecific fat fingeredness.

File: package/smt.changes [hunk 2]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Mon Dec  3 13:30:33 UTC 2018 - wstephenson@suse.com
+
+- Version 3.0.39
```
Comment (ikapelyukhin) [inline package/smt.changes:4]: Not quite 3.0.39:

https://github.com/SUSE/smt/blob/a58bcc20e2f0f2edca1f4cfcc535f6c4eb7c152c/Makefile#L1-L2

https://github.com/SUSE/smt/blob/a58bcc20e2f0f2edca1f4cfcc535f6c4eb7c152c/package/smt.spec#L19-L20

Comment (ikapelyukhin) [reply package/smt.changes:4]: But no worries, bumped up to 3.0.40 in #127.

**Pull request merged**